shared/js/background/onboarding.es6.js

```diff
@@ -0,0 +1,117 @@
+/**
+* This is only injected on duckduckgo.com to assist with user onboarding
+*/
+
+function createOnboardingCode ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    extensionId,
+    browser
+} = {}) {
+    const params = JSON.stringify({
+        isAddressBarQuery,
+        showWelcomeBanner,
+        showCounterMessaging,
+        extensionId,
+        browser
+    })
+
+    // TODO: upgrade to `chrome.scripting.executeScript` when we upgrade to manifest v3
+    // as it allows to inject a function with _arguments_. Here we simulate that in a hacky way
+    return `(${onboarding.toString().replace('__injectedJSON__', `'${params}'`)})()`
```

Comment:
This actually I **think** this could suffer from an escaping bug if the scope of these variables changed (although it appears these are very limited in scope so shouldn't). The arguments above having a single quote will break out of this escaping and be turned into real code.

You could actually just pass the stringified params into this:
```
`(${onboarding.toString())(${params})`
```

Then just change the function signature below:
```
function onboarding ({
        isAddressBarQuery,
        showWelcomeBanner,
        showCounterMessaging,
        browser,
        extensionId
    } = {}) {
```
(untested)

Reply:
Thanks, good catch this should now be fixed 👍